From 5c8f3e3a982cf9e8009a4b97ca36bf552ad2c26e Mon Sep 17 00:00:00 2001
Message-Id: <5c8f3e3a982cf9e8009a4b97ca36bf552ad2c26e.1429347747.git.chang-joon.lee@intel.com>
In-Reply-To: <58b01df1cceca63ad29c5d3026fe071bcc8e2015.1429347747.git.chang-joon.lee@intel.com>
References: <58b01df1cceca63ad29c5d3026fe071bcc8e2015.1429347747.git.chang-joon.lee@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Tue, 14 Apr 2015 11:30:37 +0530
Subject: [PATCH 2/4] SQUASHME! [VPG]: drm/i915: Correcting the PFI credit
 programming sequence

The "PFI credit resend to SSA" bit was set along with the PFI credits value
but this needs to be set after setting the PFI credits value and then
wait until this bit is cleared which is done by hardware once PFI credit
initialization request is sent.

Squash this patch to:
	Author: Vidya Srinivas <vidya.srinivas@intel.com>
	Date:   Sat Jul 5 00:44:15 2014 +0530
	Change-Id: Icb02162dcda4d3547fca649a3ae478ba8a14bcdf
	FOR_UPSTREAM [VPG]: drm/i915: Program the PFI credits

Change-Id: I8c1ac37d7d630d792fac73cf5e6b5a50825da93b
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26531
Signed-off-by: Deepak M <m.deepak@intel.com>
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |   18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index abc6fd0..7a019a4 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -8130,7 +8130,7 @@ int vlv_freq_opcode(struct drm_i915_private *dev_priv, int val)
 void program_pfi_credits(struct drm_i915_private *dev_priv, bool flag)
 {
 	int cd_clk, cz_clk;
-	int val = VGA_FAST_MODE_DISABLE | FORCE_CREDIT_RESEND;
+	int val = VGA_FAST_MODE_DISABLE;
 
 	if (!flag) {
 		I915_WRITE(GCI_CONTROL, VGA_FAST_MODE_DISABLE);
@@ -8144,19 +8144,23 @@ void program_pfi_credits(struct drm_i915_private *dev_priv, bool flag)
 			val |= PFI_CREDITS_63;
 		else if (IS_VALLEYVIEW(dev_priv->dev))
 			val |= PFI_CREDITS_15;
-		/* Disable before enabling */
-		I915_WRITE(GCI_CONTROL, VGA_FAST_MODE_DISABLE);
+
 		I915_WRITE(GCI_CONTROL, val);
 	} else {
 		if (IS_CHERRYVIEW(dev_priv->dev)) {
 			val |= PFI_CREDITS_12;
-
-			/* Disable before enabling */
-			I915_WRITE(GCI_CONTROL, VGA_FAST_MODE_DISABLE);
 			I915_WRITE(GCI_CONTROL, val);
-		} else
+		} else {
 			DRM_ERROR("cd clk < cz clk");
+			return;
+		}
 	}
+
+	val |= FORCE_CREDIT_RESEND;
+	I915_WRITE(GCI_CONTROL, val);
+
+	if (wait_for((I915_READ(GCI_CONTROL) & FORCE_CREDIT_RESEND) == 0, 100))
+		DRM_ERROR("timeout waiting for PFI credit programming\n");
 }
 
 /*
-- 
1.7.9.5

